mobile/src/main/java/org/openhab/habdroid/ui/ItemUpdateWidgetItemPickerActivity.kt

```diff
@@ -0,0 +1,36 @@
+package org.openhab.habdroid.ui
+
+import android.appwidget.AppWidgetManager
+import android.content.Intent
+import android.os.Bundle
+import org.openhab.habdroid.model.Item
+import org.openhab.habdroid.ui.homescreenwidget.ItemUpdateWidget
+
+class ItemUpdateWidgetItemPickerActivity(override var disabledMessageId: Int = 0) : AbstractItemPickerActivity() {
+    private var appWidgetId: Int = AppWidgetManager.INVALID_APPWIDGET_ID
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        retryButton.setOnClickListener {
+            loadItems()
+        }
+
+        appWidgetId = intent?.extras?.getInt(
+            AppWidgetManager.EXTRA_APPWIDGET_ID,
+            AppWidgetManager.INVALID_APPWIDGET_ID
+        ) ?: AppWidgetManager.INVALID_APPWIDGET_ID
+    }
+
+    override fun finish(item: Item, state: String, mappedState: String) {
+        val appWidgetManager: AppWidgetManager = AppWidgetManager.getInstance(this)
+        val label = if (item.label.isNullOrEmpty()) item.name else item.label
+        val data = ItemUpdateWidget.ItemUpdateWidgetData(item.name, state, label, mappedState, item.category.orEmpty())
+        ItemUpdateWidget.setupWidget(this, data, appWidgetId, appWidgetManager)
```

Comment:
Wouldn't it be more elegant to just send the broadcast to the widget provider and let that one handle the update?

Reply:
The docs say, that the configuration activity should handle the setup of the widget.
I can remove `data`  and `appWidgetManager` from the function and let `setupWidget()` fetch them.

Reply:
If the configuration activity manually triggers the widget receiver, it also handles the widget setup ;-) It's just a matter of code structure.

Reply:
I tried to send an intent to `ItemUpdateWidget` from `finish()`, but the widget is created, after `finish()`. So the widget with the `appWidgetId` isn't present at the time I send the broadcast.